lib/param_parsing/parameters.go

```diff
@@ -56,23 +61,44 @@ func GetParameters() Params {
 	logger = lib.InitLogger(params.LogLevel)
 	var err error
 	// Read configuration files
+	// TOML from Homedir
 	if tomlFileExists(params) {
 		params, err = getParamsTOML(params)
-	} else if tfSwitchFileExists(params) {
+		if err != nil {
+			logger.Fatalf("Failed to obtain settings from TOML config in home directory: %v", err)
+		}
+	}
+
+	if tfSwitchFileExists(params) {
 		params, err = GetParamsFromTfSwitch(params)
-	} else if terraformVersionFileExists(params) {
+		if err != nil {
+			logger.Fatalf("Failed to obtain settings from \".tfswitch\" file: %v", err)
+		}
+	}
+
+	if terraformVersionFileExists(params) {
 		params, err = GetParamsFromTerraformVersion(params)
-	} else if isTerraformModule(params) {
+		if err != nil {
+			logger.Fatalf("Failed to obtain settings from \".terraform-version\" file: %v", err)
+		}
+	}
+
+	if isTerraformModule(params) {
 		params, err = GetVersionFromVersionsTF(params)
-	} else if terraGruntFileExists(params) {
-		params, err = GetVersionFromTerragrunt(params)
-	} else {
-		params = GetParamsFromEnvironment(params)
+		if err != nil {
+			logger.Fatalf("Failed to obtain settings from Terraform module: %v", err)
+		}
 	}
-	if err != nil {
-		logger.Fatalf("Error parsing configuration file: %q", err)
+
+	if terraGruntFileExists(params) {
+		params, err = GetVersionFromTerragrunt(params)
+		if err != nil {
+			logger.Fatalf("Failed to obtain settings from Terragrunt configuration: %v", err)
```

Comment:
Instead of hardcoding and wording version sources, could we probably move those from `lib/param_parsing/*.go` files to a more global place and re-use?